android/src/main/java/com/swmansion/rnscreens/ScreenStackFragment.kt

```diff
@@ -287,64 +307,124 @@ class ScreenStackFragment :
-                    addUpdateListener { anim ->
-                        val animatedValue = anim.animatedValue as? Float
-                        animatedValue?.let { screen.translationY = it }
-                    }
-                }
+        val alphaAnimator = createAlphaAnimator(0f, dimmingDelegate.maxAlpha)
+        val slideAnimator = createSlideInAnimator()
 
-            animatorSet
-                .play(slideAnimator)
-                .takeIf {
-                    dimmingDelegate.willDimForDetentIndex(
-                        screen,
-                        screen.sheetInitialDetentIndex,
-                    )
-                }?.with(alphaAnimator)
-        } else {
-            val alphaAnimator =
-                ValueAnimator.ofFloat(dimmingDelegate.dimmingView.alpha, 0f).apply {
-                    addUpdateListener { anim ->
-                        val animatedValue = anim.animatedValue as? Float
-                        animatedValue?.let { dimmingDelegate.dimmingView.alpha = it }
-                    }
-                }
-            val slideAnimator =
-                ValueAnimator.ofFloat(0f, (coordinatorLayout.bottom - screen.top).toFloat()).apply {
-                    addUpdateListener { anim ->
-                        val animatedValue = anim.animatedValue as? Float
-                        animatedValue?.let { screen.translationY = it }
-                    }
+        animatorSet
+            .play(slideAnimator)
+            .takeIf {
+                dimmingDelegate.willDimForDetentIndex(
+                    screen,
+                    screen.sheetInitialDetentIndex,
+                )
+            }?.with(alphaAnimator)
+
+        attachCommonListeners(animatorSet, isEnter = true)
+
+        return animatorSet
+    }
+
+    private fun createExitAnimator(): Animator {
+        val animatorSet = AnimatorSet()
+        val dimmingDelegate = requireDimmingDelegate()
+
+        val alphaAnimator = createAlphaAnimator(dimmingDelegate.dimmingView.alpha, 0f)
+        val slideAnimator = createSlideOutAnimator()
+
+        animatorSet.play(alphaAnimator).with(slideAnimator)
+
+        attachCommonListeners(animatorSet, isEnter = false)
+
+        return animatorSet
+    }
+
+    private fun createAlphaAnimator(
+        from: Float,
+        to: Float,
+    ): ValueAnimator =
+        ValueAnimator.ofFloat(from, to).apply {
+            addUpdateListener { animator ->
+                (animator.animatedValue as? Float)?.let {
+                    requireDimmingDelegate().dimmingView.alpha = it
                 }
-            animatorSet.play(alphaAnimator).with(slideAnimator)
+            }
         }
+
+    private fun createSlideInAnimator(): ValueAnimator {
+        val startValueCallback = { _: Number? -> screen.height.toFloat() }
+        val evaluator = ExternalBoundaryValuesEvaluator(startValueCallback, { 0f })
+
+        return ValueAnimator.ofObject(evaluator, screen.height.toFloat(), 0f).apply {
```

Comment:
Reading this for the first time & I don't get why on slide in animator the start value is `height` and target value is `0`, at least passed here. I also see that there is an evaluator. 

I'll update the comment if I get this

Reply:
I wrote this code intitially, but I do not remember that. 😄

Reply:
Oh, is it because the sheet has already target position & we translate it back to the animation start position? That might be it.

Reply:
The initial sheet position is when the sheet bottom edge is equal to device's bottom edge, we're translating Y with a positive value to hide the sheet under the bottom edge of the device